src/parser/name_resolution.c

```diff
@@ -5119,14 +5158,18 @@ pt_remake_dblink_select_list (PARSER_CONTEXT * parser, PT_SPEC_INFO * class_spec
 
 	  if ((tmp = pt_mk_attr_def_node (parser, id_node, rmt_cols)) == NULL)
 	    {
-	      return ER_FAILED;
+	      error = ER_DBLINK;
+	      goto error_exit;
 	    }
 	  attr_def_node = attr_def_node ? parser_append_node (tmp, attr_def_node) : tmp;
 	}
     }
 
   dblink_table->cols = attr_def_node;
-  return NO_ERROR;
+
+error_exit:
+  parser->custom_print &= ~PT_PRINT_SUPPRESS_FOR_DBLINK;
```

Comment:
Does the original value of parser->custom_print always have the bit PT_PRINT_SUPPRESS_FOR_DBLINK ?
Can we always delete the bit as this line does?
Storing the original value and restoring it at this line, which is a frequent pattern through out the code, seems more general way of handling  parser->custom_print.

Reply:
printing the tree is called at various places through whole parser, compile, and optimize logic.
the custom_print should be kept the options to fit for purpose.